shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java

```diff
@@ -261,6 +265,9 @@ View onCreateView(
     }
     flutterSplashView.displayFlutterViewWithSplash(flutterView, host.provideSplashScreen());
 
+    Log.v(TAG, "Attaching FlutterEngine to FlutterView.");
+    flutterView.attachToFlutterEngine(flutterEngine);
```

Comment:
The main potential negative implication of this I can think of is that this is going to be blocking rendering from the Android side. So if for some reason `attachToFlutterEngine` takes a long time the user is going to see a black screen and Android may want to kill the Flutter app.

As I understand it this is a very cheap method and shouldn't block for any noticeable amount of time though, so I'm not worried about it.

Reply:
Yeah, that was the tradeoff made here. We'll accept whatever this blocking call is in exchange for not having the very noticeable black screen at each app switch. I think this blocking call was the original impetus to post it on the main handler.